doc-site/docs/architecture/distributed_sequencer.md

```diff
@@ -0,0 +1,1437 @@
+
+```proto
+message EndorsementRequest {
+    google.protobuf.Any attestation_request = 1;
+    string idempotency_key = 2;
+    string transaction_id = 3;
+    string contract_address = 4;
+    string party = 5;
+    google.protobuf.Any transaction_specification = 6;
+    repeated google.protobuf.Any verifiers = 7;
+    repeated google.protobuf.Any signatures = 8;
+    repeated google.protobuf.Any inputStates = 9;
+    repeated google.protobuf.Any readStates = 10;
+    repeated google.protobuf.Any outputStates = 11;
+    repeated google.protobuf.Any infoStates = 12;
+}
+```
+
+##### <a name="message-endorsement-response"></a>Endorsement response
+
+```proto
+message EndorsementResponse {
+    string transaction_id = 1;
+    string idempotency_key = 2;
+    string contract_address = 3;
+    optional google.protobuf.Any endorsement = 4;
+    string party = 5;
+    string attestation_request_name = 6;
+}
+```
+
+##### <a name="message-endorsement-response"></a>Endorsement error
+
+When a coordinator receives a `EndorsementError` it must remove that transaction, and any dependant transactions from the current graph of assembled transactions and queue them up for re-assembly.
+
+```proto
+message EndorsementError {
+    string transaction_id = 1;
+    string idempotency_key = 2;
+    string contract_address = 3;
+    optional string revert_reason = 4;
+    string party = 5;
+    string attestation_request_name = 6;
+}
+```
+
+#### Heartbeat
+
+Once a `coordinator` node has accepted a delegation, it will continue to periodically send `HeartbeatNotification` messages to the `sender` node for that transaction. The coordinator sends a heartbeat notification to all nodes for which it is currently coordinating transactions.  The message sent to any given node must include the ids for all active delegations. The message may contain other delegation ids that are unknown to the receiving node.
+
+This is a fire and forget notification.  There is no acknowledgement expected.
+
+```mermaid
+sequenceDiagram
+  %%{init: { "sequence": { "mirrorActors":false }}}%%
+  participant C as Coordinator
+  participant S1 as Sender 1
+  participant S2 as Sender 2
+  loop Every `HeartBeatInterval`
+    C -) S1: Heartbeat
+    C -) S2: Heartbeat
+  end
+  
+```  
+
+##### <a name="message-heartbeat-notification"></a>Heartbeat notification
+
+When a node receives a heartbeat message from a coordinator node, it should compare that notification with its record of inflight transactions and active delegations
+ - if the receiving node has any in flight transactions that have been delegated to coordinators other than the source of the received heartbeat, then the receiving node should initiate a re-evaluation of its choice of coordinator
+
+```proto
+message HeartbeatNotification {
+    string heartbeat_id = 1;
+    google.protobuf.Timestamp timestamp = 2;
+    string idempotency_key = 3;
+    string contract_address = 4;
+    repeated string transaction_ids = 5;
```

Comment:
This is great. I assume this is only the transaction IDs that have been delegated _by the sender this heartbeat is targeted to_?

Reply:
I am proposing that it may include other transactions and those should be ignored by the receiver of the message.